src/sender.rs

```diff
@@ -553,10 +556,7 @@ where
                 // this can happen for example when a device is primary, without any secondaries
                 // and we send a message to ourselves (which is only a SyncMessage { sent: ... })
                 // addressed to self
-                warn!(
-                    "no messages were encrypted: this should not really happen and most likely implies a logic error."
-                );
-                break;
+                return Err(MessageSenderError::NoMessagesToSend);
```

Comment:
As far as I understand, with the new logic from #338, this code path can't be triggerend anymore by send-to-self?

Reply:
indeed, hence why I wanna make it an error and we shall see if/when we break it.

Reply:
Sounds good! Merge ahead.